### PR TITLE
Opal 136 preview for merged kalendar

### PR DESCRIPTION
**Pull request by norvarju**
The preview displays every calendar event of the user with random colors in a swipable format. The preview activity gets displayed with 5 seconds delay due to the data retrieval from google API. When submiting preview the app navigates back to kalendar fragment with 500 milisecond delay because the backend API gets called on button click, and with no delay the newly merged calendar does not get displayed on the kalendar fragment.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -27,8 +27,16 @@
         <activity android:name=".SelectCalendarActivity" />
         <activity android:name=".SharingOptionsActivity" />
         <activity android:name=".ChooseAccountActivity" />
-        <activity android:name=".StaticWeekViewActivity" />
+
+        <activity android:name=".WeekViewActivity" />
+        <!-- <activity android:name=".StaticWeekViewActivity" /> -->
```
Comment (bekobarna) [inline app/src/main/AndroidManifest.xml]: we dont need this line

File: app/src/main/java/com/greenfox/kalendaryo/BaseActicity.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
+package com.greenfox.kalendaryo;
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/BaseActicity.java]: BaseActicity??

File: app/src/main/java/com/greenfox/kalendaryo/LoginActivity.java [hunk 3]
```diff
@@ -211,7 +211,7 @@ public void onFailure(Call<KalUser> call, Throwable t) {
                     t.printStackTrace();
                 }
             });
-            Toast.makeText(this, "Saved!", Toast.LENGTH_LONG).show();
+            //Toast.makeText(this, "Saved!", Toast.LENGTH_LONG).show();
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/LoginActivity.java]: why is it commented out?

File: app/src/main/java/com/greenfox/kalendaryo/LoginActivity.java [hunk 4]
```diff
@@ -227,7 +227,7 @@ private void editKalPref(String email, String userName, String accessToken, Stri
             kalPref.putAuth(googleAuth);
 
         }
-        Toast.makeText(this, "Sorry, you can't add the account you are already logged into!", Toast.LENGTH_LONG).show();
+        //Toast.makeText(this, "Sorry, you can't add the account you are already logged into!", Toast.LENGTH_LONG).show();
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/LoginActivity.java]: same here

File: app/src/main/java/com/greenfox/kalendaryo/StaticWeekViewActivity.java [hunk 5]
```diff
@@ -51,118 +51,6 @@ protected void onCreate(Bundle savedInstanceState) {
         event.setColor(getResources().getColor(R.color.event_color_01));
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/StaticWeekViewActivity.java]: we can remove the whole Activity if we dont use it

File: app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java [hunk 6]
```diff
@@ -20,6 +20,7 @@
 import com.greenfox.kalendaryo.models.KalPref;
 import com.greenfox.kalendaryo.models.responses.GetKalendarListResponse;
 
+
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/fragments/KalendarFragment.java]: why the empty line?

File: app/src/main/java/com/greenfox/kalendaryo/http/RetrofitClient.java [hunk 7]
```diff
@@ -16,7 +15,7 @@
 
 public class RetrofitClient {
 
-    static String BASE_URL_BACKEND = BuildConfig.LOCAL_IP_ADDRESS;
+    static String BASE_URL_BACKEND = "http://kalendaryobackenddev-env.um8evv5pid.eu-west-3.elasticbeanstalk.com/";
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/http/RetrofitClient.java]: use the default BuildConfig.LOCAL_IP_ADDRESS, not the deployed version

File: app/src/main/java/com/greenfox/kalendaryo/services/WeekViewService.java [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package com.greenfox.kalendaryo.services;
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/services/WeekViewService.java]: why the empty file?

**Review by bekobarna — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/greenfox/kalendaryo/ChooseAccountActivity.java [hunk 9]
```diff
@@ -62,34 +69,57 @@ protected void onCreate(Bundle savedInstanceState) {
...
                 new DividerItemDecoration(recyclerview.getContext(),
                         recyclerLayoutManager.getOrientation());
         recyclerview.addItemDecoration(dividerItemDecoration);
+        return recyclerview;
+    }
 
-        accountService.listAccountsFromBackend(recyclerview, false, getIntent());
+    public void launchEventService() {
+        Intent intentToEventService = new Intent(ChooseAccountActivity.this, EventService.class);
+        Bundle bundle2 = new Bundle();
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/ChooseAccountActivity.java]: maybe rename from bundle2 to something better

File: app/src/main/java/com/greenfox/kalendaryo/WeekViewActivity.java [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class WeekViewActivity extends AppCompatActivity implements
+        MonthLoader.MonthChangeListener {
+
+    private KalPref kalPref;
+    private Kalendar kalendar;
+    private GoogleApi googleApi;
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/WeekViewActivity.java:45]: we can use Dagger to have GoogleApi and BackendApi in this activity

File: app/src/main/java/com/greenfox/kalendaryo/WeekViewActivity.java [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+        kalPref = new KalPref(this.getApplicationContext());
+
+        kalendar = (Kalendar) getIntent().getSerializableExtra("list");
+
+        mWeekView = findViewById(R.id.weekview);
+        mWeekView.setMonthChangeListener(this);
+
+        sendToBackend = findViewById(R.id.send_to_backend);
+
+        sendToBackend.setOnClickListener(new View.OnClickListener() {
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/WeekViewActivity.java]: replace this onClickListener with lambda, just like in other acitivities

File: app/src/main/java/com/greenfox/kalendaryo/services/EventService.java [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import retrofit2.Response;
+
+public class EventService extends IntentService {
+
+    private GoogleApi googleApi;
```
Comment (bekobarna) [inline app/src/main/java/com/greenfox/kalendaryo/services/EventService.java]: use Dagger to inject GoogleApi

**Review by pataricza1987 — DISMISSED** [review]

**Review by bekobarna — APPROVED** [review]

**Review by pataricza1987 — APPROVED** [review]

**Pull request merged**